src/main/java/pl/simpleascoding/tutoringplatform/api/UserController.java

```diff
@@ -22,28 +23,54 @@ class UserController {
+
+        return new ResponseEntity<>(rscpDTO.body(), headers, httpStatus);
     }
 
     @GetMapping("/confirm-registration")
-    ResponseEntity<RscpDTO<?>> confirmRegistration(@RequestParam String tokenValue) {
-        return new ResponseEntity<>(userService.confirmUserRegistration(tokenValue), HttpStatus.OK);
+    ResponseEntity<?> confirmRegistration(@RequestParam String tokenValue) {
+        RscpDTO<?> rscpDTO = userService.confirmUserRegistration(tokenValue);
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("message", rscpDTO.message());
+        HttpStatus httpStatus = HttpStatus.resolve(rscpDTO.status().value());
+
+        return new ResponseEntity<>(rscpDTO.body(), headers, httpStatus);
     }
 
     @PostMapping("/change-password")
-    ResponseEntity<RscpDTO<?>> changeUserPassword(@RequestBody ChangeUserPasswordDTO dto, Principal principal, HttpServletRequest request) {
-        return new ResponseEntity<>(userService.changeUserPassword(dto, principal.getName(), request.getRequestURL().toString()), HttpStatus.OK);
+    ResponseEntity<?> changeUserPassword(@RequestBody ChangeUserPasswordDTO dto,
+                                                  Principal principal, HttpServletRequest request) {
+        RscpDTO<?> rscpDTO = userService.changeUserPassword(dto, principal.getName(), request.getRequestURL().toString());
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("message", rscpDTO.message());
+        HttpStatus httpStatus = HttpStatus.resolve(rscpDTO.status().value());
+
+        return new ResponseEntity<>(rscpDTO.body(), headers, httpStatus);
     }
 
     @GetMapping("/confirm-change-password")
-    ResponseEntity<RscpDTO<?>> confirmChangeUserPassword(@RequestParam String tokenValue) {
-        return new ResponseEntity<>(userService.confirmChangeUserPassword(tokenValue), HttpStatus.OK);
+    ResponseEntity<?> confirmChangeUserPassword(@RequestParam String tokenValue) {
+        RscpDTO<?> rscpDTO = userService.confirmChangeUserPassword(tokenValue);
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("message", rscpDTO.message());
+        HttpStatus httpStatus = HttpStatus.resolve(rscpDTO.status().value());
+
+        return new ResponseEntity<>(rscpDTO.body(), headers, httpStatus);
     }
 
     @PatchMapping
-    ResponseEntity<RscpDTO<UserDTO>> modifyUser(@RequestBody ModifyUserDTO dto, Principal principal){
-        return new ResponseEntity<>(userService.modifyUser(dto, principal.getName()), HttpStatus.OK);
+    ResponseEntity<UserDTO> modifyUser(@RequestBody ModifyUserDTO dto, Principal principal) {
+        RscpDTO<UserDTO> rscpDTO = userService.modifyUser(dto, principal.getName());
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("message", rscpDTO.message());
+        HttpStatus httpStatus = HttpStatus.resolve(rscpDTO.status().value());
+
+        return new ResponseEntity<>(rscpDTO.body(), headers, httpStatus);
```

Comment:
You decided to split all ResponseEntity parameters on varriable-value line:
```java
        HttpHeaders headers = new HttpHeaders();
        headers.add("message", rscpDTO.message());
        HttpStatus httpStatus = HttpStatus.resolve(rscpDTO.status().value());

        return new ResponseEntity<>(rscpDTO.body(), headers, httpStatus);
```
very good idea. So maybe it's good to keep this convetion to the end?
Instead:
```java
	rscpDTO.body()
```
Should be in this particular case:
```java
	UserDTO = rscpDTO.body();
```
	
And simlar with all prev methods.